### PR TITLE
SPUI-655/478 > Add 2fa UI

### PR DESCRIPTION
**Pull request by VovaMedvediev**

### CODE REVIEW

File: lib/shared/features/two_fa/two_fa_phone/view/two_fa_phone.dart [hunk 1]
```diff
@@ -74,60 +75,63 @@ class TwoFaPhone extends HookWidget {
...
+                      await twoFaN.verifyCode();
+                    },
+                    pinError: pinError.value,
                   ),
-                ),
-                if (twoFa.union is Loading) const ScaffoldLoader(),
-              ],
+                  const SpaceH7(),
+                  if (timer != 0 && !twoFa.showResend)
+                    ResendInText(text: 'You can resend in $timer seconds')
```
Comment (eli1stark) [inline lib/shared/features/two_fa/two_fa_phone/view/two_fa_phone.dart:114]: comma

File: lib/shared/features/two_fa/two_fa_phone/view/two_fa_phone.dart [hunk 2]
```diff
@@ -74,60 +75,63 @@ class TwoFaPhone extends HookWidget {
...
+                  else ...[
+                    const ResendInText(text: "Didn't receive the code?"),
```
Comment (eli1stark) [inline lib/shared/features/two_fa/two_fa_phone/view/two_fa_phone.dart:116]: comma

File: lib/shared/components/texts/verification_description_text.dart [hunk 3]
```diff
@@ -14,21 +15,18 @@ class VerificationDescriptionText extends HookWidget {
...
         children: [
           TextSpan(
             text: text,
           ),
           TextSpan(
             text: boldText,
-            style: const TextStyle(
-              fontWeight: FontWeight.bold,
-            ),
+            style: sBodyText1Style.copyWith(color: colors.black),
```
Comment (eli1stark) [inline lib/shared/components/texts/verification_description_text.dart:29]: comma

File: lib/shared/components/texts/verification_description_text.dart [hunk 4]
```diff
@@ -14,21 +15,18 @@ class VerificationDescriptionText extends HookWidget {
...
   Widget build(BuildContext context) {
+    final colors = useProvider(sColorPod);
+
     return RichText(
       text: TextSpan(
-        style: TextStyle(
-          fontSize: 16.sp,
-          color: Colors.black,
-        ),
+        style: sBodyText1Style.copyWith(color: colors.grey1),
```
Comment (eli1stark) [inline lib/shared/components/texts/verification_description_text.dart:22]: comma

**Review by eli1stark — CHANGES_REQUESTED** [review]

File: lib/shared/features/two_fa/two_fa_phone/view/two_fa_phone.dart (hunk 1 above)
Comment (VovaMedvediev) [reply lib/shared/features/two_fa/two_fa_phone/view/two_fa_phone.dart]: done

File: lib/shared/features/two_fa/two_fa_phone/view/two_fa_phone.dart (hunk 2 above)
Comment (VovaMedvediev) [reply lib/shared/features/two_fa/two_fa_phone/view/two_fa_phone.dart]: done

File: lib/shared/components/texts/verification_description_text.dart (hunk 3 above)
Comment (VovaMedvediev) [reply lib/shared/components/texts/verification_description_text.dart]: done

File: lib/shared/components/texts/verification_description_text.dart (hunk 4 above)
Comment (VovaMedvediev) [reply lib/shared/components/texts/verification_description_text.dart]: done

**Pull request merged**